python/lsst/ip/isr/isrTask.py

```diff
@@ -1981,16 +1981,15 @@ def darkCorrection(self, exposure, darkExposure, invert=False):
         expScale = exposure.getInfo().getVisitInfo().getDarkTime()
         if math.isnan(expScale):
             raise RuntimeError("Exposure darktime is NAN.")
-        if darkExposure.getInfo().getVisitInfo() is not None:
+        if darkExposure.getInfo().getVisitInfo() is not None \
+                and not math.isnan(darkExposure.getInfo().getVisitInfo().getDarkTime()):
```

Comment:
Do we get NaN because of a bug? Or because pipe_drivers/cp_pipe isn't putting that information anywhere?

Reply:
I'm pretty sure it's the latter: we've been treating calibrations as `Image` and/or `MaskedImage` mostly, so their VisitInfo was always null. Now we're reading them as Exposure, so they get a default-constructed VisitInfo.